src/Promitor.Core.Scraping/Configuration/Model/Metrics/PrometheusMetricDefinition.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections.Generic;
+
+namespace Promitor.Core.Scraping.Configuration.Model.Metrics
+{
+    public class PrometheusMetricDefinition
+    {
+        public PrometheusMetricDefinition(string name, string description, Dictionary<string, string> labels)
+        {
```

Comment:
So this is slightly tricky - it goes back to the way the validation stuff works where it's currently done against the runtime model, and it expects a null name to be allowed (presumably so that we can display a slightly friendlier message on startup?).

If I add the guards in a load of the tests under `Validation.Metrics.ResourceTypes` start failing.

I could maybe try to change the validation code to work against the config model rather than the runtime model, but I wanna hold off until I hear your thoughts on it.

Reply:
I think the current approach is ok to work against the runtime model and validate that as that's what really matters, thanks for checking!

Resolving.